src/libs/opts/opts.c

```diff
@@ -1297,6 +1807,124 @@ void setOption (Key * option, const char * value, bool repeated)
 	}
 }
 
+/**
+ * Writes the options from parseArgs into proc keys
+ */
+int writeOptions (Key * command, Key * commandKey, Key * commandArgs, bool writeArgs, bool * argsWritten, KeySet * options,
+		  struct Specification * spec, KeySet * ks, const char * progname, const char ** envp, Key * parentKey)
+{
+	Key * helpKey = keyNew (keyName (command), KEY_END);
+	keyAddName (helpKey, "/long/help");
+	if (ksLookup (options, helpKey, KDB_O_DEL) != NULL)
+	{
+		// show help
+		char * lastSlash = strrchr (progname, '/');
+		if (lastSlash != NULL)
+		{
+			progname = lastSlash + 1;
+		}
+
+		char * usage = generateUsageLine (progname, ksLookup (spec->commands, command, 0), commandArgs);
+		char * optionsText = generateOptionsList (spec->keys, commandKey);
+		char * commandsText = generateCommandsList (spec->keys, commandKey);
+		char * argsText = generateArgsList (spec->keys, commandKey);
+		char * envsText = generateEnvsList (spec->keys);
+
+		keySetMeta (parentKey, "internal/libopts/help/usage", usage);
+		keySetMeta (parentKey, "internal/libopts/help/options", optionsText);
+		keySetMeta (parentKey, "internal/libopts/help/commands", commandsText);
+		keySetMeta (parentKey, "internal/libopts/help/args", argsText);
+		keySetMeta (parentKey, "internal/libopts/help/envs", envsText);
+
+		elektraFree (usage);
+		elektraFree (optionsText);
+		elektraFree (commandsText);
+		elektraFree (argsText);
+		elektraFree (envsText);
+		return 1;
+	}
+
+	KeySet * envValues = parseEnvp (envp);
+
+	Key * argsParent = keyNew (keyName (command), KEY_END);
+	keyAddBaseName (argsParent, "args");
+	KeySet * args = elektraArrayGet (argsParent, options);
+	keyDel (argsParent);
+
+	Key * keyWithOpt;
+	ksRewind (spec->keys);
+	while ((keyWithOpt = ksNext (spec->keys)) != NULL)
```

Comment:
I'm not even sure that would work, but even if it works, it would be very confusing, since there are two different `result` variables used in different context.

I will instead add comments to the `continue`s and `return`s to clarify what is happening.

Reply:
Thank you!